lib/utils/utils.js

```diff
@@ -261,13 +269,19 @@ function getExternalContractUrl(file) {
   } else {
     return null;
   }
-  const match = url.match(
-    /\.[a-z]+\/([-a-zA-Z0-9@:%_+.~#?&\/=]+)/
-  );
-  return {
-    url,
-    filePath: constants.httpContractsDirectory + match[1]
-  };
+  if (file.startsWith('ipfs')) {
+    const match = file.match(/ipfs:\/\/([-a-zA-Z0-9]+)\/(.*)/);
+    return {
+      url,
+      filePath: constants.httpContractsDirectory + match[1] + '/' + match[2]
+    };
```

Comment:
This can be put after line 246. Basically, it's the same `if`, so you can return from that `if`.

Reply:
Hi @jrainville, sorry for the late reply. Yes sure, sorry for that.

Reply:
Hi @jrainville , I have just committed these changes and i have deleted the `.sol` file `SimpleStorageWithIPFSImport`. I included the ipfs import file in `SimpleStorageWithHttpImport.sol`, It is missing only the change for retrieving ipfs url.Thanks.